### DISCUSSION TITLE
Sync fills up system drive

### DISCUSSION THREAD

**Discussion by mhdask** [abraunegg/onedrive#2508] [Q&A]
Hello
I  have installed onedrive on nixos, and have configured the sync_dir to be on another drive, than my system drive.
The config itself lies on the systemdrive.

When i run onedrive --synchronization --resync my system drive fills up, and every other file download starts to fail.
Is there anyway this can be fixed, without having to move the config files?

**Comment by abraunegg** [comment]
> When i run onedrive --synchronization --resync my system drive fills up, and every other file download starts to fail.

Regardless of your configuration - this is odd, as there are protection mechanisms in place to prevent this.

Lets start with the basics. What is the output of these 2 commands:

* `onedrive --version`
* `onedrive --display-config`

**Comment by mhdask** [comment]
```bash
$ onedrive --version
onedrive v2.4.25
```
and
```bash
$ onedrive --display-config
Configuration file successfully loaded
onedrive version                             = v2.4.25
Config path                                  = /home/mdask/.config/onedrive
Config file found in config path             = true
Config option 'sync_dir'                     = /home/mdask/mnt/backups/onedrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive/
Config option 'disable_notifications'        = false
Config option 'min_notify_changes'           = 5
Config option 'skip_dir'                     =
Config option 'skip_dir_strict_match'        = false
Config option 'skip_file'                    = ~*|.~*|*.tmp
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 600
Config option 'monitor_log_frequency'        = 6
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = true
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = false
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               =
Config option 'azure_ad_endpoint'            =
Config option 'azure_tenant_id'              = common
Config option 'user_agent'                   =
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 600
Config option 'ip_protocol_version'          = 0
Config option 'sync_root_files'              = false
Selective sync 'sync_list' configured        = false
Config option 'sync_business_shared_folders' = false
Business Shared Folders configured           = false
Config option 'webhook_enabled'              = false
```

**Comment by mhdask** [comment]
Okay, i found the problem - It was actually cloned to my home directory, which i guess i specified by the "~".. How would i get it to syncrhonize to this path: /mnt/backup/onedrive

**Reply by abraunegg** [reply]
Use the specic path, and check with `--display-config`

Once happy - use `--dry-run` to test.

Please read the documentation as well.

**Reply by mhdask** [reply]
I changed sync_dir to "/mnt/backup/onedrive/" but i get the following error:
```bash
ERROR: Unable to create local OneDrive syncDir - /mnt/backups: Permission denied
```

**Reply by abraunegg** [reply]
@mhdask 
You have a permission problem.

Please do some research and learning on this - this is not a problem this client can solve. Some suggested search terms for you to try:
* [how to solve unix directory permission problems](https://www.google.com/search?q=how+to+solve+unix+directory+permission+problems&rlz=1C1SQJL_enAU887AU887&oq=how+to+solve+unix+directory+permission+problems&aqs=chrome..69i57j33i160.11623j0j7&sourceid=chrome&ie=UTF-8)
* [how to fix unix directory permission problems](https://www.google.com/search?q=how+to+fix+unix+directory+permission+problems&sca_esv=571601278&rlz=1C1SQJL_enAU887AU887&sxsrf=AM9HkKm4_ilGjjgHg1Bjd-3ZUTWSLoZjjA%3A1696704367850&ei=b6chZY7KM9qlhwO8taqABA&ved=0ahUKEwjOvYmpzOSBAxXa0mEKHbyaCkAQ4dUDCBA&uact=5&oq=how+to+fix+unix+directory+permission+problems&gs_lp=Egxnd3Mtd2l6LXNlcnAiLWhvdyB0byBmaXggdW5peCBkaXJlY3RvcnkgcGVybWlzc2lvbiBwcm9ibGVtczIIECEYoAEYwwQyCBAhGKABGMMESLqdAVCsmQFYw5sBcAZ4AZABAJgBvQGgAesDqgEDMC4zuAEDyAEA-AEBwgIKEAAYRxjWBBiwA-IDBBgAIEGIBgGQBgg&sclient=gws-wiz-serp)

You will need to spend some time learning the problem, learning how to solve the problem and then fix your problem.

The root cause is:
* You are running the 'onedrive' client as a certain user
* That user does not have access to create the directory in that location you have chosen

To solve this you need to:
* Change your directory permissions to allow the create to occur
* Create the folder manually, assign the correct permissions to that folder so that the 'onedrive' client is run, it has the correct permissions to that location.
